src/mumble/MainWindow.h

```diff
@@ -178,6 +180,10 @@ class MainWindow : public QMainWindow, public Ui::MainWindow {
 
 	Search::SearchDialog *m_searchDialog = nullptr;
 
+	MenuLabel *m_localVolumeLabel;
+	UserLocalVolumeSlider *m_userLocalVolumeSlider;
+	ListenerLocalVolumeSlider *m_listenerLocalVolumeSlider;
```

Comment:
Should we use smart-pointers for these, instead? That'd seem like a bit more modern to me :thinking: 
For QObjects, you can use the std smart-pointers (I'd go with `unique_ptr` here), but you have to use the custom deleter function from
https://github.com/mumble-voip/mumble/blob/8fc359ea3fafbbb136abece6d422af409274b23e/src/QtUtils.h#L25
which calls `deleteLater` rather than `delete` on the object (the latter can sometimes let Qt crash).

Reply:
Are you 100% sure unique_ptr are compatible with the Qt parent/child architecture? I am no Qt expert, but what I have found suggest that this is generally something that you do not want to do [see this](https://stackoverflow.com/questions/59634605/using-stdunique-ptr-on-qt)

I also could not find any indication that this was done with Qt objects anywhere else in the Mumble code base

Reply:
When the objects you note here are created in MainWindow.cpp they are passed the Window as parent

Reply:
Yes it's compatible as long as you use deleteLater instead delete on them once freeing.

Qt's parent/child mechanism is quite horrible tbh. I mean it works and all but it just produces massively smelly code 👀

Reply:
But, how do I do that? The deletion is handled by the parent. We create only one Instance of the sliders which are set to the  correct listener/user on click. We do not explicitly delete these QWidgets... like ever...

Reply:
Since you create them in the class's constructor, the place to delete them would be the destructor. But the thing is: Smart-pointers take care of that automatically - there is no need to explicitly delete these objects. So the end result will be very similar to what Qt does in its own "hacky" way but using standard C++ features, instead.

Reply:
Ok yes, I get the first part. But what you suggest would require to not use the parent/child relationship for the sliders. If I use unique_ptr AND give the sliders a parent, there might be a double pointer delete scenario... So do you want me to not pass MainWindow::this as parent to the sliders?

Reply:
Maybe I am missing a point here, because I am not that familiar with Qt

Reply:
No there won't be a double delete. Qt will notice that these items have already been deleted